src/adagrams.js

```diff
@@ -1,8 +1,101 @@
-const Adagrams = {
-  drawLetters() {
-    // Implement this method for wave 1
-  },
-};
+class Adagrams {  
+  static drawLetters() {
+    const letterDistribution = {
+      A: 9, B: 2, C: 2, D: 4, E: 12, F: 2, G: 3, H: 2, I: 9, J: 1, K: 1, L: 4, M: 2,
+      N: 6, O: 8, P: 2, Q: 1, R: 6, S: 4, T: 6, U: 4, V: 2, W: 2, X: 1, Y: 2, Z: 1
+    }
+
+    const allLetters = []
+
+    for (let letter in letterDistribution) {
+      let i = 0;
+      while (i < letterDistribution[letter]) {
+        allLetters.push(letter);
+        i += 1;
+      }
+    }
+
+    // https://css-tricks.com/snippets/javascript/shuffle-array/
+    allLetters.sort(function() { return 0.5 - Math.random() });
+
+    return allLetters.slice(0, 10);
+  }
+
+  static usesAvailableLetters(input, lettersInHand) {
+    let validWord = true;
+    let userWord = input.toUpperCase().split("");
+
+    userWord.forEach((char) => {
+      let index = lettersInHand.indexOf(char);
+
+      if (index === -1) {
+        validWord = false;
+        return;
+      } else {
+        lettersInHand.splice(index, 1)
```

Comment:
Consider making a deep copy of `lettersInHand` so that invoking `usesAvailableLetters` does not change `lettersInHand`